connector_ecommerce/unit/sale_order_onchange.py

```diff
@@ -207,38 +161,36 @@ def _play_line_onchange(self, line, previous_lines, order):
         :return: the value of the sale order updated with the onchange result
         :rtype: dict
         """
-        sale_line_model = self.session.pool.get('sale.order.line')
-
+        line_model = self.env['sale.order.line']
         # Play product_id onchange
         args, kwargs = self._get_product_id_onchange_param(line,
                                                            previous_lines,
                                                            order)
-        res = sale_line_model.product_id_change(self.session.cr,
-                                                self.session.uid,
-                                                *args,
-                                                **kwargs)
-        # TODO refactor this with merge_values
-        vals = res.get('value', {})
-        for key in vals:
-            if key not in line:
-                if sale_line_model._columns[key]._type == 'many2many':
-                    line[key] = [(6, 0, vals[key])]
-                else:
-                    line[key] = vals[key]
+        context = kwargs.pop('context', {})
+        values = line_model.with_context(context).product_id_change(*args,
+                                                                    **kwargs)
+        self.merge_values(line, values)
```

Comment:
`merge_values` does not handle the `many2many` case properly.
For instance, the `product_id_change` method returns only a list of integers for `tax_id`. These have to be transformed into many2many syntax, otherwise the order lines will not have taxes.